### DISCUSSION TITLE
TTGO_ESP_POE-il9341 error during build

### DISCUSSION THREAD

**Discussion by bensuffolk** [HASwitchPlate/openHASP#341] [General]
I am looking to add Ethernet to the Lanbon L8 [#65] and thought it would be sensible to view another module with ethernet first, to see how it works, but it seems there are a number of errors when trying to build this one.

I will try and work through them to see if I can figure out the issues. 

The first one I can see is that wifi_run_scripts() is not defined if HASP_USE_WIFI = 0. This is related to the scripts to run on connection / disconnection. I guess the other errors will relate to similar additions that have been added over time to functionality.

**Comment by fvanroie** [comment]
Ethernet builds have not been actively maintained as there wasn't really a demand for it... so far.

I think it will be better suited to move `void wifi_run_scripts()` to `void network_run_scripts(bool online)` in `hasp_network.cpp`.
That's supposed to be the general network code for both Wifi/Ethernet.

I'll have a look...

Do you actually own this board to test with?

**Reply by bensuffolk** [reply]
No I don't. Just saw it was about the only one with Ethernet in it, so thought it would be a good start to see if the code base compiled.
